mst peerDep is outdated

When running `npm i` this error occurs:
```
PS C:\Users\alexa\vermehrung> npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: vermehrung.ch@1.14.10
npm ERR! Found: mobx-state-tree@5.1.8
npm ERR! node_modules/mobx-state-tree
npm ERR!   mobx-state-tree@"5.1.8" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer mobx-state-tree@"^3.2.1" from mst-persist@0.1.3
npm ERR! node_modules/mst-persist
npm ERR!   mst-persist@"0.1.3" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! C:\Users\alexa\AppData\Local\npm-cache\_logs\2023-07-12T13_22_49_699Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\alexa\AppData\Local\npm-cache\_logs\2023-07-12T13_22_49_699Z-debug-0.log
```

This forces to install using `npm i --force` which is not recommended.

As no one will be using mobx-state-tree@^3.2.1 any more, it would help to update the peer dependency to a newer version to `Fix the upstream dependency conflict`.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -1,5 +1,5 @@
 name: CI
-on: [push]
+on: [push, pull_request]
 jobs:
   ci:
     name: CI - Node ${{ matrix.node-version }}, ${{ matrix.os }}

--- a/package.json
+++ b/package.json
@@ -37,7 +37,7 @@
     "changelog": "changelog-maker"
   },
   "peerDependencies": {
-    "mobx-state-tree": "^3.2.1"
+    "mobx-state-tree": "^5.1.8"
   },
   "dependencies": {},
   "devDependencies": {

--- a/package.json
+++ b/package.json
@@ -43,8 +43,8 @@
   "devDependencies": {
     "@agilgur5/changelog-maker": "^3.0.0",
     "jest-without-globals": "^0.0.2",
-    "mobx": "^5.11.0",
-    "mobx-state-tree": "^3.14.0",
+    "mobx": "^6.3.0",
+    "mobx-state-tree": "^5.1.8",
     "tsdx": "github:agilgur5/tsdx#dist",
     "typescript": "^3.5.2"
   }

--- a/package.json
+++ b/package.json
@@ -29,7 +29,7 @@
   "scripts": {
     "start": "tsdx watch",
     "build": "tsdx build",
-    "tsc": "tsc --noEmit",
+    "tsc": "tsc --noEmit --skipLibCheck",
     "test": "tsdx test",
     "test:coverage": "tsdx test --coverage",
     "test:pub": "npm run build && npm pack",

--- a/package.json
+++ b/package.json
@@ -37,7 +37,7 @@
     "changelog": "changelog-maker"
   },
   "peerDependencies": {
-    "mobx-state-tree": "^5.1.8"
+    "mobx-state-tree": ">=3.2.1"
   },
   "dependencies": {},
   "devDependencies": {

